src/app/pages-block/pages-block.component.ts

```diff
@@ -1,33 +1,51 @@
 import { Component, OnInit } from '@angular/core';
 import { CoursePage } from '../interfaces/classes';
+import { FoundCoursesPipe } from './pipes/found-courses.pipe';
+import { CoursesService } from './services/courses.service';
+
 
 @Component({
   selector: 'app-pages-block',
   templateUrl: './pages-block.component.html',
-  styleUrls: ['./pages-block.component.scss']
+  styleUrls: ['./pages-block.component.scss'],
+  providers: [CoursesService]
 })
+
 export class PagesBlockComponent implements OnInit {
-  courses: CoursePage[];
 
-  loadNewCourses() {
-    console.log('here is come action');
+  courses: CoursePage[] = [];
+
+  constructor(private coursesPages: CoursesService) {
+    this.courses = coursesPages.getCoursesList();
```

Comment:
everytime try to avoid do something inside constructor
you have OnInit hook for most cases
within constructor we write only if it's only way